RCS1261: "Resource can be disposed asynchronously" does not take lock statement into account

**Product and Version Used**: 

Roslynator.Analyzers 4.9.0

**Steps to Reproduce**:

Consider the following code:

```cs
object fileStreamLock = new();

await SomeAsyncMethod();

async Task SomeAsyncMethod()
{
	lock (fileStreamLock)
	{
		using FileStream fs = new("test.txt", FileMode.OpenOrCreate);
		// Do something...
	}

	await Task.Yield(); // For demonstration purposes
}
```

`RCS1261` is reported, saying that the `FileStream` can be disposed asynchronously, but doing so results in [compiler error CS1996](https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/compiler-messages/cs1996) (Cannot 'await' in the body of a 'lock' statement). This is not valid C#:

```cs
lock (fileStreamLock)
{
        await using FileStream fs = new("test.txt", FileMode.OpenOrCreate);
}
```

--- a/ChangeLog.md
+++ b/ChangeLog.md
@@ -23,6 +23,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 ### Fixed
 
 - Fix analyzer [RCS1055](https://josefpihrt.github.io/docs/roslynator/analyzers/RCS1055) ([PR](https://github.com/dotnet/roslynator/pull/1361))
+- Fix analyzer [RCS1261](https://josefpihrt.github.io/docs/roslynator/analyzers/RCS1261) ([PR](https://github.com/dotnet/roslynator/pull/1374))
 
 ## [4.9.0] - 2024-01-10
 

--- a/src/Analyzers/CSharp/Analysis/DisposeResourceAsynchronouslyAnalyzer.cs
+++ b/src/Analyzers/CSharp/Analysis/DisposeResourceAsynchronouslyAnalyzer.cs
@@ -107,6 +107,10 @@ public sealed class DisposeResourceAsynchronouslyAnalyzer : BaseDiagnosticAnalyz
                 Analyze(context, anonymousMethod.Modifiers, usingKeyword, anonymousMethod);
                 break;
             }
+            else if (node is LockStatementSyntax)
+            {
+                return;
+            }
         }
     }
 

--- a/src/Tests/Analyzers.Tests/RCS1261DisposeResourceAsynchronouslyTests.cs
+++ b/src/Tests/Analyzers.Tests/RCS1261DisposeResourceAsynchronouslyTests.cs
@@ -921,4 +921,28 @@ internal class Disposable : IDisposable, IAsyncDisposable
 }
 ");
     }
+
+    [Fact, Trait(Traits.Analyzer, DiagnosticIdentifiers.DisposeResourceAsynchronously)]
+    public async Task TestNoDiagnostic_LockStatement()
+    {
+        await VerifyNoDiagnosticAsync(@"
+using System.IO;
+using System.Threading.Tasks;
+
+abstract class C
+{
+    object _lock = new();
+
+    async Task Foo()
+    {
+        lock (_lock)
+        {
+            using FileStream fs = new(""test.txt"", FileMode.OpenOrCreate);
+        }
+
+        await Task.Yield();
+    }
+}
+");
+    }
 }

